### ISSUE TITLE
docker commit > invalid tar header

### ISSUE THREAD

**Issue by SeidChr** [moby/moby#33586]
**Description**

Hi,

After importing a couple of databases in the ms developer sql database image, then stopping it, and committing, i get the following error:
```
PS > docker commit ContainerXY my/tagError 
response from daemon: re-exec error: exit status 1: output: archive/tar: invalid tar header
```
The Databases are quite big though, but anyway, it should not fail like that.
Is there any more info i can give you to solve that issue?

**Steps to reproduce the issue:**
1. Start ms developer sql database image
2. import 120gig of databases (final findows filter directory size)
3. commit the image

**Describe the results you received:**
```
response from daemon: re-exec error: exit status 1: output: archive/tar: invalid tar header
```

**Describe the results you expected:**
image comitted correctly

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version` & `docker info`:**

```
PS C:\Windows\system32> docker info
Containers: 2
 Running: 0
 Paused: 0
 Stopped: 2
Images: 209
Server Version: 17.03.1-ee-3
Storage Driver: windowsfilter
 Windows: 
Logging Driver: json-file
Plugins: 
 Volume: local
 Network: l2bridge l2tunnel nat null overlay transparent
Swarm: inactive
Default Isolation: process
Kernel Version: 10.0 14393 (14393.1198.amd64fre.rs1_release_sec.170427-1353)
Operating System: Windows Server 2016 Standard
OSType: windows
Architecture: x86_64
CPUs: 12
Total Memory: 63.92 GiB
Name: S6
ID: 2OZY:DSZI:JEUU:4ISW:GJ23:F2TA:F5U3:XR4Z:YZKU:SAID:362W:MMXD
Docker Root Dir: D:\System\Data\Docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: -1
 Goroutines: 22
 System Time: 2017-06-02T09:56:38.4659153+02:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

PS C:\Windows\system32> docker version
Client:
 Version:      17.03.1-ee-3
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   3fcee33
 Built:        Thu Mar 30 19:31:22 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.03.1-ee-3
 API version:  1.27 (minimum version 1.24)
 Go version:   go1.7.5
 Git commit:   3fcee33
 Built:        Thu Mar 30 19:31:22 2017
 OS/Arch:      windows/amd64
 Experimental: false

PS C:\Windows\system32> 
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Windows Server 2016 and updated docker
https://social.msdn.microsoft.com/Forums/en-US/14d2ad0c-1116-42c7-8449-29d495be4f30/docker-commit-invalid-tar-header?forum=windowscontainers
Screenshot of windowsfilter folder properties https://social.msdn.microsoft.com/Forums/getfile/1080800

**Comment by snorrk** [comment]
I'm hitting the same issue, it seems to be related to the size of the container.

**Steps**

$containerName = "Test01"

docker run -d --name $containerName  --volume c:\shared:c:\shared --storage-opt size=60G --memory 10g microsoft/iis

$containerInfo = docker container inspect $containerName | ConvertFrom-Json
Enter-PSSession -ContainerID $containerInfo.id -RunAsAdministrator

copy c:\shared\18-gigabyte-file.dat c:\

exit

docker stop container $containerName
docker commit $containerName testcontainerimage

**Result**

Error response from daemon: re-exec error: exit status 1: output: archive/tar: invalid tar header

**Environment**

_docker version_
Client:
 Version:      17.06.2-ee-3
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   915cbaa
 Built:        Thu Sep 21 23:01:20 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.06.2-ee-3
 API version:  1.30 (minimum version 1.24)
 Go version:   go1.8.3
 Git commit:   915cbaa
 Built:        Thu Sep 21 23:10:04 2017
 OS/Arch:      windows/amd64
 Experimental: false

_docker info_
Containers: 4
 Running: 2
 Paused: 0
 Stopped: 2
Images: 3
Server Version: 17.06.2-ee-3
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: l2bridge l2tunnel nat null overlay transparent
 Log: awslogs etwlogs fluentd json-file logentries splunk syslog
Swarm: inactive
Default Isolation: process
Kernel Version: 10.0 14393 (14393.1715.amd64fre.rs1_release_inmarket.170906-1810)
Operating System: Windows Server 2016 Datacenter
OSType: windows
Architecture: x86_64
CPUs: 4
Total Memory: 16GiB
Name: snbeck-test01
ID: Z2FF:YVBZ:BI7Y:FD2Y:2LY4:46VW:FVYT:UAD5:57WS:UQOF:2IT4:YAQC
Docker Root Dir: C:\ProgramData\docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

**Comment by thaJeztah** [comment]
@n4ss @stevvooe related to https://github.com/moby/moby/issues/35075, or different code path?

**Comment by JanneRantala** [comment]
Same issue here with Windows 10 Creators Update and latest Docker from edge channel. I was trying to restore 1.5 gig database file to SQL Server when got that error.

Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:09 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.24)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:50:27 2017
 OS/Arch:      windows/amd64
 Experimental: true

**Comment by thaJeztah** [comment]
ping @johnstep PTAL

**Comment by snorrk** [comment]
For what it's worth, I also had SQL Server and database files in my container.

**Comment by dmarkle** [comment]
I can repro this on a SQL 2017 (microsoft/mssql-server-windows-developer) restoring a ~30GB database.  I'm seeing this particular error manifest itself when I `docker-compose build` my test environment.  The RUN for the database restore works fine and I get the standard SQL Server message saying that I've processed 3763487 pages,, but once that's done, I get 
ERROR: Service '<NAME>' failed to build: re-exec error: exit status 1: output : archive/tar: invalid tar header

**Comment by thaJeztah** [comment]
Does it work if you set the `size` storage option for the daemon to a bigger value? (I think the default may be 20 or 30G); https://github.com/docker/cli/blob/master/docs/reference/commandline/dockerd.md

Easiest way to do so is to create a `daemon.json` configurationfile for this (see the linked document); typing from my phone so can't easily give a full example, but happy to help.

After changing the configuration, the dockerd daemon must be restarted

**Comment by dmarkle** [comment]
No.  I had to set the size storage option and restart dockerd to even get
to this point.  If I hadn’t made the storage-opt adjustment, I would have
just received an error restoring the database complaining that there was
not enough space on disk.

**Comment by thaJeztah** [comment]
Thanks for the extra information; I was wondering if the "invalid tar header" would have been due to that (and just an incorrect / confusing error)

**Comment by mikebolland** [comment]
We are also experiencing this issue dockerising oracle and windows server core as the base image

 ---> Running in a2e00d5102f8
Copying database files
1% complete
3% complete
11% complete
18% complete
26% complete
33% complete
37% complete
Creating and starting Oracle instance
40% complete
45% complete
50% complete
55% complete
56% complete
60% complete
62% complete
Completing Database Creation
66% complete
70% complete
73% complete
85% complete
96% complete
100% complete
re-exec error: exit status 1: output: archive/tar: invalid tar header

**Comment by thaJeztah** [comment]
@johnstep I think you mentioned that this could possibly be due to https://github.com/Microsoft/go-winio using an older version of archive/tar (based on Go 1.6)? https://github.com/Microsoft/go-winio/tree/424d62c72ec0db3fa2b19791b6f96c7eb5a5e412/archive/tar

**Comment by johnstep** [comment]
Yes, apparently the 8 GB restriction was removed in Go 1.8. We need to review whether that modified copy of archive/tar in go-winio is still required and, if so, update it; otherwise, remove it and switch to the standard runtime package. I spoke with @jhowardmsft separately, as well.

**Comment by mikebolland** [comment]
I notice most of the posts above have remapped docker to d:\ (in my case symlinked c:\programdata to d:\programdata) so i have tested this on docker-ee 17.9 with a very large c:\ partition to confirm windows mklink is not the cause

Sadly the result is the same
re-exec error: exit status 1: output: archive/tar: invalid tar header

**Comment by mikebolland** [comment]
Is the attempt to fix this being done as part of https://github.com/moby/moby/pull/35739 ?

**Comment by ElasticCoder** [comment]
I have the same problem. Restoring a 15GB database into SQLExpress.

```
RESTORE DATABASE successfully processed 839885 pages in 32.105 seconds (204.379 MB/sec).
re-exec error: exit status 1: output: archive/tar: invalid tar header
```

```
PS C:\Source Code\GitHub\Backoffice-container> docker image ls
REPOSITORY                               TAG                 IMAGE ID            CREATED             SIZE
<none>                                   <none>              ebaf1a726468        3 minutes ago       15.2GB
```

**Comment by danter** [comment]
I made some tests on my machine with the different ouput types (STDOUT or file output switch -i) and the sizes differs with a lot. When I then opened the files in a diff-tool it complains about different carriage return types. Moving on I also opened the two files in a hex editor, and turns out the version that was created using STDOUT had a hex 0x00 character for every second character. My guess is that it might be the classical problem where carriage return differs between linux and windows, and also a problem where windows uses UTF-16 by default and not UTF-8. Effectively corrupting any form of character strings or binary data.

**Comment by TheHackmeister** [comment]
Just wanted to drop a line saying I'm experiencing this issue in 18.03 stable. I'm trying to import a DB that ends up taking 25ish GB. It run fine (once I increased storage-opts), but as soon as I try to save the container to an image, I get the invalid tar header error. This is the case when I build build the container, as well as when I manually run the steps in a container and then trying to commit said container. 

It sounds like we have a possible solution, but if you need any more information or help, I'm game to experiment and would love to get this working. Just let me know how I can help.

Windows 10: 1607 (Build 14393.2125)

```
C:\inetpub\wwwroot>docker version
Client:
 Version:       18.03.0-ce
 API version:   1.37
 Go version:    go1.9.4
 Git commit:    0520e24
 Built: Wed Mar 21 23:06:28 2018
 OS/Arch:       windows/amd64
 Experimental:  false
 Orchestrator:  swarm

Server:
 Engine:
  Version:      18.03.0-ce
  API version:  1.37 (minimum version 1.24)
  Go version:   go1.9.4
  Git commit:   0520e24
  Built:        Wed Mar 21 23:21:06 2018
  OS/Arch:      windows/amd64
  Experimental: false
```

```
C:\inetpub\wwwroot>docker info
Containers: 4
 Running: 2
 Paused: 0
 Stopped: 2
Images: 42
Server Version: 18.03.0-ce
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: ics l2bridge l2tunnel nat null overlay transparent
 Log: awslogs etwlogs fluentd gelf json-file logentries splunk syslog
Swarm: inactive
Default Isolation: hyperv
Kernel Version: 10.0 14393 (14393.2125.amd64fre.rs1_release.180301-2139)
Operating System: Windows 10 Enterprise
OSType: windows
Architecture: x86_64
CPUs: 4
Total Memory: 15.66GiB
Name: shachmeist1217
ID: 3GO5:EVIM:P2QZ:7PGM:VNSV:7GTK:MOWX:NHP2:43GP:6DNM:57IN:MXGR
Docker Root Dir: C:\ProgramData\Docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: -1
 Goroutines: 41
 System Time: 2018-04-23T11:47:18.7785299-06:00
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Registry Mirrors:
 https://peakdockerregistry.azurecr.io/
Live Restore Enabled: false
```

**Comment by smasherprog** [comment]
I have this problem as well, is this going to be fixed at all?

**Comment by devteamexpress** [comment]
Same here

**Comment by TBBle** [comment]
I suspect this was fixed in #41430, assuming https://github.com/moby/moby/issues/33586#issuecomment-354991916 was an accurate diagnosis.

**Comment by thaJeztah** [comment]
Ah, yes, I guess this was fixed by https://github.com/moby/moby/pull/41430 / https://github.com/microsoft/go-winio/pull/175

Let me go ahead and close
